producer/src/accounts/accounts.go

```diff
@@ -0,0 +1,243 @@
+
+	c.Signature, _ = sender.Sign(rand.Reader, preHash, nil)
+	c.SigLen = uint8(len(c.Signature))
+}
+
+/*
+Check balance (ideal scenario):
+Open table
+Get hash of contract
+Verify signature with hash and public key
+Go to table and find sender
+Confirm balance is sufficient
+Update Account Balances (S & R)		// ony updated when true
+Increment Table Nonce
+
+1. verify signature
+hashed contract = sha 256 hash ( version + spubkey + rpubkeyhash + value + nonce )
+verify (hashed contract, spubkey, signature) (T)
+2. validate amount
+check table to see if sender's balance >= contract amount (T)
+3. validate nonce
+check to see that nonce is 1 + table nonce for that account (T)
+
+If all 3 are true, update table
+*/
+func ValidateContract(c Contract, tableName string) bool {
+	table, err := sql.Open("sqlite3", tableName)
+	if err != nil {
+		//"Failed to open sqlite3 table"
+		return false
+	}
+	defer table.Close()
+
+	spubkey := keys.EncodePublicKey(&c.SenderPubKey)
+	preSerial := make([]byte, 374)
+
+	binary.LittleEndian.PutUint16(preSerial[0:2], c.Version)   // 2
+	copy(preSerial[2:180], spubkey)                            //178
+	copy(preSerial[180:212], c.RecipPubKeyHash)                //32
+	binary.LittleEndian.PutUint64(preSerial[212:220], c.Value) //8
+	binary.LittleEndian.PutUint64(preSerial[220:228], c.Nonce) //8
+
+	hashedContract := block.HashSHA256(preSerial)
+
+	// stores r and s values needed for ecdsa.Verify
+	var esig struct {
+		R, S *big.Int
+	}
+	if _, err := asn1.Unmarshal(c.Signature, &esig); err != nil {
+		fmt.Println(err)
+	}
+
+	// if the ecdsa.Verify is true then check the rest of the contract against whats in the database
+	if ecdsa.Verify(&c.SenderPubKey, hashedContract, esig.R, esig.S) {
+		rows, err := table.Query("SELECT public_key_hash , balance, nonce FROM account_balances")
+		if err != nil {
+			fmt.Println("Failed to create rows to look for public key")
+		}
+		defer rows.Close()
+
+		// look for the public key that pertains to the contract and verify its balance and nonce
+		var pkh string
+		var tblBal int
+		var tblNonce int
+		for rows.Next() {
+			rows.Scan(&pkh, &tblBal, &tblNonce)
+			if reflect.DeepEqual(pkh, (hex.EncodeToString(block.HashSHA256(keys.EncodePublicKey(&c.SenderPubKey))))) {
+				if tblBal >= int(c.Value) {
+					if tblNonce+1 == int(c.Nonce) {
+						rows.Close()
+						c.UpdateAccountBalanceTable(tableName)
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+/*
+spkh = sha256 ( serialized sender pub key )
+find sender public key hash
+decrease value from sender public key hash account
+increment their nonce by one
+increase value of recipient public key hash account by contract value
+*/
+func (c *Contract) UpdateAccountBalanceTable(table string) {
+	tbl, err := sql.Open("sqlite3", table)
+	if err != nil {
+		fmt.Println("Failed to open sqlite3 table")
+	}
+	defer tbl.Close()
+
+	rows, err := tbl.Query("SELECT public_key_hash , balance, nonce FROM account_balances")
```

Comment:
Can you use a `WHERE public_key_hash = ?` here so that you won't need to loop through each item later?